src/NServiceBus.Core/EndpointConfiguration.cs

```diff
@@ -30,13 +30,15 @@ public EndpointConfiguration()
             : base(new SettingsHolder())
         {
             configurationSourceToUse = new DefaultConfigurationSource();
-
             pipelineCollection = new PipelineConfiguration();
             Settings.Set<PipelineConfiguration>(pipelineCollection);
             Pipeline = new PipelineSettings(pipelineCollection.MainPipeline);
 
             Settings.Set<QueueBindings>(new QueueBindings());
 
+            instanceIdGenerationStrategy = new EndpointInstanceIdGenerationStrategy();
+            Settings.Set<EndpointInstanceIdGenerationStrategy>(instanceIdGenerationStrategy);
```

Comment:
I don't think so. It is similar to setting the `PipelineConfiguration` instance -- using settings as a service locator.